### DISCUSSION TITLE
Type Marshalling

### DISCUSSION THREAD

**Discussion by SergiyYermak** [mullerj/norm#3] [General]
| typedef | C++ | C# |
|-|-|-|
|  | unsigned int | uint |
|  | unsigned int* | ref uint |
|  | unsigned int& | ref uint |
|  | unsigned short | ushort |
|  | unsigned long | ulong |
|  | unsigned char | byte |
|  | char* | string |
|  | const char* | in string |
|  | const void* | in nuint |
| NormInstanceHandle | const void* | in nuint |
| NormSessionHandle | const void* | in nuint |
| NormNodeHandle | const void* | in nuint |
| NormObjectHandle | const void* | in nuint |
| NormNodeId | uint32_t | uint |
| UINT32 | uint32_t | uint |
| UINT16 | uint16_t | ushort |
| UINT16* | uint16_t* | ref ushort |
| UINT16& | uint16_t& | ref ushort |
| UINT8 | uint8_t | byte |
|  | size_t | nuint |
|  | custom_type* | ref custom_type |

**Comment by mullerj** [comment]
It looks like unmanaged type void* will map to managed type nuint.  It probably still needs in since it has const.

**Comment by mullerj** [comment]
size_t should probably map to nuint, but it is platform dependent: https://stackoverflow.com/questions/772531/net-equivalent-of-size-t
